spec/firebase_app_distribution_action_spec.rb

```diff
@@ -1,131 +1,193 @@
-            })
-          )
-        Fastlane::Actions::FirebaseAppDistributionAction.upload_binary("app_id", "binary_path")
+    it 'should upload the binary successfully' do
+      stubs.post("/app-binary-uploads?app_id=app_id", "Hello World") do |env|
+        expect(env.url.path).to eq("/app-binary-uploads")
+        [
+          202,
+          {},
+          {
+            token: "projects/project_id/apps/app_id/releases/-/binaries/binary_hash"
+          }
+        ]
       end
+      Fastlane::Actions::FirebaseAppDistributionAction.upload_binary("app_id", "binary_path")
     end
 
-    context 'when testing with invalid parameters' do
-      it 'should crash if given an invalid app_id' do
-        expect(fake_connection).to receive(:post)
-          .with("/app-binary-uploads?app_id=invalid_app_id", "Hello World")
-          .and_raise(Faraday::ResourceNotFound.new("404"))
-        expect { Fastlane::Actions::FirebaseAppDistributionAction.upload_binary("invalid_app_id", "binary_path") }
-          .to raise_error(ErrorMessage::INVALID_APP_ID)
+    it 'should crash if given an invalid app_id' do
+      stubs.post("/app-binary-uploads?app_id=invalid_app_id", "Hello World") do |env|
+        expect(env.url.path).to eq("/app-binary-uploads")
+        [
+          404,
+          {},
+          {}
+        ]
       end
+      expect { Fastlane::Actions::FirebaseAppDistributionAction.upload_binary("invalid_app_id", "binary_path") }
+        .to raise_error("#{ErrorMessage::INVALID_APP_ID}: invalid_app_id")
+    end
 
-      it 'should crash if given an invalid binary_path' do
-        expect(File).to receive(:open)
-          .with("invalid_binary_path")
-          .and_raise(Errno::ENOENT.new("file not found"))
-        expect { Fastlane::Actions::FirebaseAppDistributionAction.upload_binary("app_id", "invalid_binary_path") }
-          .to raise_error(ErrorMessage::APK_NOT_FOUND)
-      end
+    it 'should crash if given an invalid binary_path' do
+      expect(File).to receive(:open)
+        .with("invalid_binary_path")
+        .and_raise(Errno::ENOENT.new("file not found"))
+      expect { Fastlane::Actions::FirebaseAppDistributionAction.upload_binary("app_id", "invalid_binary_path") }
+        .to raise_error("#{ErrorMessage::APK_NOT_FOUND}: invalid_binary_path")
     end
   end
 
   describe '#upload' do
     # Empty for now
   end
 
+  describe '#post_notes' do
+    it 'should post the notes successfully' do
+      stubs.post("/v1alpha/apps/app_id/releases/release_id/notes", "{\"releaseNotes\":{\"releaseNotes\":\"release_notes\"}}") do |env|
+        expect(env.url.path).to eq("/v1alpha/apps/app_id/releases/release_id/notes")
+        [
+          200,
+          {},
+          {}
+        ]
+      end
+      Fastlane::Actions::FirebaseAppDistributionAction.post_notes("app_id", "release_id", "release_notes")
+    end
+
+    it 'should not post if release notes are empty' do
+      expect(stubs).not_to(receive(:post))
```

Comment:
Should you be expecting `conn` to not receive post? My impression is that the stubs object is only used from the test, but I could be wrong.

Reply:
Yes, fixed